p2p/src/lib.rs

```diff
@@ -38,14 +39,34 @@ pub enum Recipients<P: PublicKey> {
 }
 
 /// Interface for sending messages to a set of recipients.
-pub trait Sender: Clone + Debug + Send + Sync + 'static {
+pub trait Sender: Debug + Clone + Send + Sync + 'static {
     /// Error that can occur when sending a message.
     type Error: Debug + StdError + Send + Sync;
 
     /// Public key type used to identify recipients.
     type PublicKey: PublicKey;
 
-    /// Send a message to a set of recipients.
+    /// Sends a message to a set of recipients.
+    ///
+    /// # Offline Recipients
+    ///
+    /// If a recipient is offline at the time a message is sent, the message
+    /// will be dropped. It is up to the application to handle retries (if
+    /// necessary).
+    ///
+    /// # Rate Limiting
```

Comment:
As discussed offline, I can't imagine a p2p implementation (even if not fully connected) that doesn't have some notion of rate limiting. Am ok keeping this a top-level comment.